src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -368,6 +368,49 @@ namespace JSONHelpers
         SaveFancyZonesData();
     }
 
+    bool FancyZonesData::IsAnotherWindowOfApplicationInstanceZoned(HWND window) const
+    {
+        std::scoped_lock lock{ dataLock };
+        auto processPath = get_process_path(window);
+        if (!processPath.empty())
+        {
+            auto history = appZoneHistoryMap.find(processPath);
+            if (history != appZoneHistoryMap.end())
+            {
+                DWORD processId = 0;
+                GetWindowThreadProcessId(window, &processId);
+
+                if (history->second.processIdToHandleMap.find(processId) == history->second.processIdToHandleMap.end())
```

Comment:
Redundant calls to find/at, it would be much easier to extract an iterator and then use it instead.